### PR TITLE
[Scala 3] Add tests for formatting trait parameters, local type declaration and new type wildcard symbol

### PR DESCRIPTION
**Pull request by tgodzik**
~~Scala 3 added support for specifying trait constructor parameters, which seems to work out of the box. I added some tests just in case.~~

~~EDIT: Seems that it's also working fine in case of local type declarations, that can now be defined inside a local block. Added tests for that also.~~ 

It seems that a number of new features work out of the box in scalafmt so I added tests that check:
- local type declaration
- new wildcard symbol `?`
- trait parameters

I decided to put everything into single PR instead of each in a separate one, because it seems all those are small features/changes.

### CODE REVIEW

File: scalafmt-tests/src/test/resources/dotty/Type.stat [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+>>>
+def hello() = {
+  type T[
+      AAAAAAAAAAAAA <: BBBBBBBBBBBBBB[
+        ?
+      ]
+  ]
+  val a: T
+}
```
Comment (kitbellew) [inline scalafmt-tests/src/test/resources/dotty/Type.stat:44]: missing LF

**Review by olafurpg — APPROVED** [review]
LGTM 👍 This is safe to merge IMO given it has no code changes

**Review by kitbellew — CHANGES_REQUESTED** [review]
I looked over the codebase looking to make sure that (`Token.KwTrait`, `Token.KwClass`) and (`Defn.Class`, `Defn.Trait`) are always used together.

There is at least example when this is not true: `FormatOps.verticalMultiline`.

File: scalafmt-tests/src/test/resources/dotty/Type.stat (hunk 1 above)
Comment (tgodzik) [reply scalafmt-tests/src/test/resources/dotty/Type.stat]: Fixed!

**Comment by tgodzik** [comment]
> I looked over the codebase looking to make sure that (`Token.KwTrait`, `Token.KwClass`) and (`Defn.Class`, `Defn.Trait`) are always used together.
> 
> There is at least example when this is not true: `FormatOps.verticalMultiline`.

I added it there and added some more tests for verticalMultiline. I haven't really been able to find a test case that would fail without the change though.

**Review by kitbellew — APPROVED** [review]

**Pull request merged**